pkg/window/strategy/fixed/fixed.go

```diff
@@ -23,34 +23,153 @@ limitations under the License.
 package fixed
 
 import (
+	"container/list"
+	"github.com/numaproj/numaflow/pkg/window/keyed"
+	"sync"
 	"time"
 
 	"github.com/numaproj/numaflow/pkg/window"
 )
 
 // Fixed implements Fixed window.
+// Fixed maintains the state of active windows
+// All the operations in Fixed (see window.Windower) order the entries in the ascending order of start time.
+// So the earliest window is at the front and the oldest window is at the end.
 type Fixed struct {
 	// Length is the temporal length of the window.
 	Length time.Duration
+	// entries is the list of active windows that are currently being tracked.
+	// windows are sorted in chronological order with the earliest window at the head of the list.
+	// list.List is implemented as a doubly linked list which allows us to traverse the nodes in
+	// both the directions.
+	// Although the worst case time complexity is O(n), because of the time based ordering and
+	// since the elements are rarely out of order, the amortized complexity works out to be closer to O(1)
+	// Because most of the keys are expected to be associated with the most recent window, we always start
+	// the traversal from the tail of the list for Get and Create Operations. For Remove Operations, since
+	// the earlier windows are expected to be closed before the more recent ones, we start the traversal
+	// from the Head.
+	entries *list.List
+	lock    sync.RWMutex
 }
 
 var _ window.Windower = (*Fixed)(nil)
 
-// NewFixed returns a Fixed window.
+// NewFixed returns a Fixed windower.
 func NewFixed(length time.Duration) *Fixed {
```

Comment:
As per the comment, you are returning `windower`; in that case, it should be`AlignedWindow`.

Reply:
done.